Update for dates ( > 1600 days to < 30 )

cplpearce opened the pull request:
Previous epoch times were >1600 days ago.

from `1461113959088` > Wednesday, April 20, 2016 12:59:19.088 AM
to     `1597896001234` > Thursday, August 20, 2020 4:00:01.234 AM

While this functioned correctly, quite a few cohort mates mentioned it threw off their calculations and looked unusual.

cplpearce committed:
Update for dates ( > 1600 days to < 30 )

Previous epoch times were >1600 days ago.

from `1461113959088` > Wednesday, April 20, 2016 12:59:19.088 AM
to     `1597896001234` > Thursday, August 20, 2020 4:00:01.234 AM

While this functioned correctly, quite a few cohort mates mentioned it threw off their calculations and looked unusual.

andydlindsay commented:
Thanks for this! And I agree that students are often confused when they see a 4 year old date. 

My concern with this solution is that it will need to be done again (and again). If we move to a more dynamic solution, then we shouldn't have to revisit this.

cplpearce commented:
~~I was wondering about that - since you're already using [user-helper.js](https://github.com/lighthouse-labs/tweeter/blob/master/server/lib/util/user-helper.js), ideally it could be expanded to write some random tweets to the [initial-tweets.json](https://github.com/lighthouse-labs/tweeter/blob/master/server/data-files/initial-tweets.json) file upon running the server?  Since it's already a self made [faker](https://www.npmjs.com/package/faker)-esque function and it handles everything but date already it wouldn't be a huge leap to add a quick fs function.~~

~~I'd like to but I wanted to make sure there's no issues about a student editing core functions of the project.  If it's not a problem I'd like to take a stab at it.~~

I'm guessing any major changes would not jive with the tutorial data - I guess it's as simple as a quick fs.write

cplpearce committed:
Add tweet date update function

index.js now requires dateAdjust and calls it immediately.

cplpearce committed:
Create date-adjust.js

date-adjust.js

Called upon running index.js, date-adjust will synchronously update the initial-tweets.json created_at epoch values. 

The values will be changed to 1 and 2 days before Date.now()

cplpearce commented:
Done, the new function in lib will update the tweets on each run.  

It will synchronously read the initial-tweets, change the dates, and re-write them then continue starting the express server.

cplpearce committed:
Fix dayDecrement() structure

The dayDecrement function previously had a declarative return, changed to a declaration and then return.

cplpearce committed:
Change comments

Clarified comments relating to new tweets, and path description

cplpearce committed:
Change filename to better fit js conventions

cplpearce committed:
Change function dayBefore name to adjustedDate

cplpearce committed:
Fix bad upload.

An old version of the initial-tweets was uploaded by mistake.  Fixed.

cplpearce committed:
Update date-adjust to write tweets oldest first.

Originally order of tweets was wrong, fixed date-adjust so the newest tweets appear at the bottom.

srsgores reviewed (changes requested):
Let's refactor out the magic numbers.

cplpearce committed:
Refactored date-adjust for more concise logic

cplpearce committed:
Fix JSON data read, add dynamic range to date modifications

cplpearce reviewed (commented):
All requested changes have been made.

cplpearce committed:
Revert to original

cplpearce committed:
Revert to original

cplpearce committed:
Revert to original

cplpearce reviewed (commented):
All requested changes have been made.

cplpearce committed:
Clarify magic numbers in date-adjust and expand comments in index.js

cplpearce committed:
Change tweetsJSON to a let declaration, and declare it so it's more clear the orig array is being mutated

cplpearce committed:
Update comment regarding IIFE and remove declaration

cplpearce commented:
Changes made, all requests are now satisfied <sup>(I think)<sup>.

cplpearce committed:
Working change regarding IIFE's
